### PR TITLE
Add generate test duration file cron

### PR DESCRIPTION
**Pull request by safoinme**
## Describe changes
I implemented/fixed _ to achieve _.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

## Summary by CodeRabbit

- **Chores**
	- Improved continuous integration workflow to include test duration tracking across different environments.

### CODE REVIEW

**Comment by strickvl** [comment]
@coderabbitai review

File: .github/workflows/ci-fast.yml [hunk 1]
```diff
@@ -86,3 +86,16 @@ jobs:
       python-version: ${{ matrix.python-version }}
       test_environment: ${{ matrix.test_environment }}
     secrets: inherit
+  test-generate-test-duration:
+    strategy:
+      matrix:
+        os: [ubuntu-dind-runners]
+        python-version: ['3.10']
+        test_environment: [default]
```
Comment (strickvl) [inline .github/workflows/ci-fast.yml:94]: does this need a matrix? can't we just use the OS + Python version in the `with` params below directly?

File: .github/workflows/generate-test-duration.yml [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+              git commit -m "new changes on test duration file detected at $(date +'%Y%m%d')"
+              git push origin $FEATURE_BRANCH
+          else
+              echo "No changes in .test_duration"
+          fi
+      - name: Create PR if changes found
+        if: success()
+        run: |
+          if [ -n "$FEATURE_BRANCH" ]; then
+              gh pr create --base develop --head $FEATURE_BRANCH --title "Update `.test_duration` due to detected changes" --body "This PR was automatically created by the CI pipeline. Please review and merge if you agree."
```
Comment (strickvl) [inline .github/workflows/generate-test-duration.yml:208]: Please add at least one of us on the team as a reviewer in this command. Also add the 'tests' and 'internal' tags using the command. (There are flags available for this).

**Review by strickvl — CHANGES_REQUESTED** [review]
Looks ok, but this isn't a `cron` job. It just runs every time we run the fast CI which feels maybe too often?

File: .github/workflows/generate-test-duration.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+          # to fix problems with the fork() calls (see this thread
+          # for more information: http://sealiesoftware.com/blog/archive/2017/6/5/Objective-C_and_fork_in_macOS_1013.html)
+      OBJC_DISABLE_INITIALIZE_FORK_SAFETY: 'YES'
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v4.1.1
+        with:
+          ref: feature/OSSK-365-cron-test-duration-file-generate
```
Comment (strickvl) [inline .github/workflows/generate-test-duration.yml:27]: just for testing?

Comment (safoinme) [reply .github/workflows/generate-test-duration.yml:27]: yes, will change it once this works

File: .github/workflows/generate-test-duration.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
+---
+name: Generate test duration file
+on:
+  workflow_call:
+  schedule:
+    - cron: 0 8 * * 1  # Run every Monday at 8 am
```
Comment (avishniakov) [inline .github/workflows/generate-test-duration.yml:6]: Will new tests just be randomly assigned somewhere if not found in this file?

**Review by avishniakov — APPROVED** [review]

File: .github/workflows/generate-test-duration.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            # Ubuntu integration tests run as 6 shards
+        run: |
+          bash scripts/test-coverage-xml.sh "" default "" "" store-durations
+      - name: Check difference in .test_durations
+        run: |
+          git diff --quiet || echo "Changes found"
+          if [ -n "$(git status --porcelain)" ]; then
+            # Commit changes directly to the specified branch
+            git add .test_durations
+            git commit -m "new changes on test duration file detected at $(date +'%Y%m%d-%H%M%S')"
```
Comment (strickvl) [inline .github/workflows/generate-test-duration.yml:47]:
```suggestion
            git commit -m "Update with new changes for test duration file at $(date +'%Y%m%d-%H%M%S')"
```

**Review by strickvl — CHANGES_REQUESTED** [review]

**Review by strickvl — APPROVED** [review]

**Pull request merged**